src/main/groovy/liquibase/harness/compatibility/advanced/AdvancedTest.groovy

```diff
@@ -0,0 +1,188 @@
+import static liquibase.harness.util.FileUtils.*
+import static liquibase.harness.util.SnapshotHelpers.snapshotMatchesSpecifiedStructure
+import static liquibase.harness.util.TestUtils.*
+
+class AdvancedTest extends Specification{
+    @Shared
+    RollbackStrategy strategy;
+    @Shared
+    List<DatabaseUnderTest> databases;
+    String resourcesDirFullPath = System.getProperty("user.dir") + "/src/test/resources/"
+    String resourcesDirPath = "/src/test/resources/"
+
+    def setupSpec() {
+        databases = TestConfig.instance.getFilteredDatabasesUnderTest()
+        strategy = chooseRollbackStrategy()
+        strategy.prepareForRollback(databases)
+    }
+
+    @Unroll
+    def "apply end2end test for #testInput.change against #testInput.databaseName #testInput.version"() {
+        given: "read input data for advanced test"
+        String expectedSql = parseQuery(getSqlFileContent(testInput.change, testInput.databaseName, testInput.version,
+                "liquibase/harness/compatibility/advanced/expectedSql/setup"))
+
+        Map<String, Object> argsMapPrimary = new HashMap()
+        argsMapPrimary.put("url", testInput.url)
+        argsMapPrimary.put("username", testInput.username)
+        argsMapPrimary.put("password", testInput.password)
+        argsMapPrimary.put("changelogFile", testInput.primarySetupChangelogPath)
+        argsMapPrimary.put("snapshotFormat", "json")
+        argsMapPrimary.put("excludeObjects", "(?i)posts, (?i)authors, (?i)databasechangelog, (?i)databasechangeloglock")//excluding static test-harness objects
+
+        Map<String, Object> argsMapSecondary = new HashMap()
+        argsMapSecondary.put("url", testInput.referenceUrl)
+        argsMapSecondary.put("username", testInput.username)
+        argsMapSecondary.put("password", testInput.password)
+        argsMapSecondary.put("referenceUrl", testInput.url)
+        argsMapSecondary.put("referenceUsername", testInput.username)
+        argsMapSecondary.put("referencePassword", testInput.password)
+        argsMapSecondary.put("excludeObjects", "(?i)posts, (?i)authors, (?i)databasechangelog, (?i)databasechangeloglock")
+        boolean shouldRunChangeSet
+
+        and: "check database under test is online"
+        def connection = testInput.database.getConnection()
+        shouldRunChangeSet = connection instanceof JdbcConnection
+        assert shouldRunChangeSet: "Database ${testInput.databaseName} ${testInput.version} is offline!"
+
+        and: "ignore testcase if it's invalid for this combination of db type and/or version"
+        shouldRunChangeSet = !expectedSql?.toLowerCase()?.contains("invalid test")
+        Assume.assumeTrue(expectedSql, shouldRunChangeSet)
+
+        and: "configuring test data for generateChangelog command for all files format"
+        def shortDbName = getShortDatabaseName(testInput.databaseName)
+        def generateChangelogMap = configureChangelogMap(testInput.generateCLResourcesPath, shortDbName)
```

Comment:
Please rename 'generateCLResourcesPath' to 'generateChangelogResourcesPath' for better understanding.